rpc/utils.go

```diff
@@ -0,0 +1,458 @@
+		if er != nil {
+			return nil, er
+		}
+		rpcTx, err := s.txResultToRPCTransaction(txResult)
+		if err != nil {
+			return nil, err
+		}
+		txs = append(txs, *rpcTx)
+	}
+
+	return txs, nil
+}
+
+// blockToRPCBlock converts a block protobuf to the RPC block type
+func (s *rpcServer) blockToRPCBlock(protoBlock *coreTypes.Block) (*Block, error) {
+	txs, err := s.txProtoBytesToRPCTransactions(protoBlock.GetTransactions())
+	if err != nil {
+		return nil, err
+	}
+
+	qc := new(conTypes.QuorumCertificate)
+	if err := codec.GetCodec().Unmarshal(protoBlock.BlockHeader.GetQuorumCertificate(), qc); err != nil {
+		return nil, err
+	}
+	partialSigs := make([]PartialSignature, 0)
+	for _, sig := range qc.GetThresholdSignature().GetSignatures() {
+		ps := PartialSignature{
+			Signature: hex.EncodeToString(sig.GetSignature()),
+			Address:   sig.GetAddress(),
+		}
+		partialSigs = append(partialSigs, ps)
+	}
+
+	qcTxs := make([]string, 0)
+	for _, txBz := range qc.GetBlock().GetTransactions() {
+		tx := base64.StdEncoding.EncodeToString(txBz)
+		qcTxs = append(qcTxs, tx)
+	}
+
+	qcBlockBz, err := codec.GetCodec().Marshal(qc.GetBlock())
+	if err != nil {
+		return nil, err
+	}
+	qcBlock := base64.StdEncoding.EncodeToString(qcBlockBz)
+
+	return &Block{
+		BlockHeader: BlockHeader{
+			Height:        int64(protoBlock.BlockHeader.GetHeight()),
+			NetworkId:     protoBlock.BlockHeader.GetNetworkId(),
+			StateHash:     protoBlock.BlockHeader.GetStateHash(),
+			PrevStateHash: protoBlock.BlockHeader.GetPrevStateHash(),
+			ProposerAddr:  hex.EncodeToString(protoBlock.BlockHeader.GetProposerAddress()),
+			QuorumCert: QuorumCertificate{
+				Height: int64(qc.GetHeight()),
+				Round:  int64(qc.GetRound()),
+				Step:   qc.GetStep().String(),
+				Block:  qcBlock,
+				ThresholdSig: ThresholdSignature{
+					Signatures: partialSigs,
+				},
+				Transactions: qcTxs,
+			},
+			Timestamp: protoBlock.BlockHeader.GetTimestampt().AsTime().String(),
+		},
+		Transactions: txs,
+	}, nil
+}
+
+// protocolActorToRPCActorTypeEnum converts a protocol actor type to the rpc actor type enum
+func protocolActorToRPCActorTypeEnum(protocolActorType coreTypes.ActorType) ActorTypesEnum {
+	switch protocolActorType {
+	case coreTypes.ActorType_ACTOR_TYPE_APP:
+		return Application
+	case coreTypes.ActorType_ACTOR_TYPE_FISH:
+		return Fisherman
+	case coreTypes.ActorType_ACTOR_TYPE_SERVICER:
+		return Servicer
+	case coreTypes.ActorType_ACTOR_TYPE_VAL:
+		return Validator
+	default:
+		panic("invalid actor type")
+	}
+}
+
+// getProtocolActorGetter returns the correct protocol actor getter function based on the actor type parameter
+func getProtocolActorGetter(persistenceContext modules.PersistenceReadContext, params GetV1P2pStakedActorsAddressBookParams) func(height int64) ([]*coreTypes.Actor, error) {
+	var protocolActorGetter = persistenceContext.GetAllStakedActors
+	if params.ActorType == nil {
+		return persistenceContext.GetAllStakedActors
+	}
```

Comment:
Do you need this if structure at all given how you set it up?

If you add a `default` to the switch statement, you can probably just do 

```
switch ActorTye {
	App:
		return GetAllApps
	Fisher:
		...
	default:
		return GetAllStakedActors
}
```

Reply:
I didn't make this one but have updated nonetheless 👍🏼